src/GitHub.App/Models/PullRequestModel.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Globalization;
+using GitHub.Primitives;
+using GitHub.VisualStudio.Helpers;
+using NullGuard;
+
+namespace GitHub.Models
+{
+    public sealed class PullRequestModel : NotificationAwareObject, IPullRequestModel
+    {
+        public PullRequestModel(int number, string title, IAccount author, DateTimeOffset createdAt, DateTimeOffset? updatedAt = null)
+        {
+            Number = number;
+            Title = title;
+            Author = author;
+            CreatedAt = createdAt;
+            UpdatedAt = updatedAt ?? CreatedAt;
+        }
+
+        public void CopyFrom(IPullRequestModel other)
```

Comment:
Yeah, makes sense for this implementation to check that the id matches. I've pushed a fix for that.

In this case the copy is only copying what it wants updated, yes. In other cases the copy might be a full deep copy, it's for the implementation to decide. I've documented the interface to clarify this.